pyvista/plotting/charts.py

```diff
@@ -3957,6 +3960,22 @@ def __init__(self, figure=None, size=(1, 1), loc=(0, 0)):
 
         self._redraw()
 
+    def deep_clean(self):
```

Comment:
Should this be private? Also, we should consider protecting the import of `matplotlib` if it's not installed.

Reply:
It should be callable by `pyvista.close_all` (and then up the `deep_clean` chain plotter->renderers->renderer->charts->chart). I just followed 'the convention' used in these other objects, which all kept it a public method.
Also the matplotlib check is part of the ChartMPL's `__init__` function, so I thought it would not be necessary here. But I can add it here as well.

Reply:
> Also the matplotlib check is part of the ChartMPL's...

No worries then. This does need to be a public method if being called externally.